### PR TITLE
Text renderer with support for colors

### PR DESCRIPTION
**Pull request by jprashanth**

### CODE REVIEW

File: libopenage/engine.cpp [hunk 1]
```diff
@@ -222,6 +222,8 @@ Engine::Engine(util::Dir *data_dir, const char *windowtitle)
 	bind_player_switch(input::action_t::SWITCH_TO_PLAYER_6, 6);
 	bind_player_switch(input::action_t::SWITCH_TO_PLAYER_7, 7);
 	bind_player_switch(input::action_t::SWITCH_TO_PLAYER_8, 8);
+
+	this->text_renderer.reset(new renderer::TextRenderer());
```
Comment (TheJJ) [inline libopenage/engine.cpp]: `this->text_renderer = std::make_unique<renderer::TextRenderer>()`

**Comment by TheJJ** [comment]
Nice work! We can merge it soon, i'll integrate it into #287 then but it is functionaly standalone.

File: libopenage/renderer/text_renderer.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	void render();
+
+private:
+	struct TextRenderBatchPass {
+		int x;
+		int y;
+		std::string text;
+	};
+
+	struct TextRenderBatch {
```
Comment (TheJJ) [inline libopenage/renderer/text_renderer.h]: One small thing left: struct names are lowercase-underscored. `text_render_batch` and `text_render_batch_pass`.

Would also be nice if you added some `/** documentation */` for the members of this class.

File: libopenage/renderer/color.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+	uint8_t r;
+	uint8_t g;
+	uint8_t b;
+	uint8_t a;
+
+};
+
+bool operator==(const Color &left, const Color &right);
+bool operator!=(const Color &left, const Color &right);
```
Comment (TheJJ) [inline libopenage/renderer/color.h]: These should also be members ov the Color class:
`class Color { bool operator==(const Color &other); };`

File: libopenage/engine.cpp [hunk 4]
```diff
@@ -278,22 +280,28 @@ void Engine::end_game() {
...
 	);
 
+	this->text_renderer->set_color(renderer::Color{0, 255, 0, 255});
 	// Draw version string in the lower left corner
 	this->render_text(
 		{5, 35}, 20,
 		"openage %s", config::version
 	);
+
+	this->text_renderer->set_color(renderer::Color{0, 0, 255, 255});
```
Comment (TheJJ) [inline libopenage/engine.cpp]: I'm not sure if we should color the on-screen texts all differently :smile:, leaving them white might be a bit better for readability.

File: libopenage/renderer/text_renderer.cpp [hunk 5]
```diff
@@ -0,0 +1,114 @@
+// Copyright 2015-2015 the openage authors. See copying.md for legal info.
+
+#include "text_renderer.h"
+
+#include <algorithm>
+#include <epoxy/gl.h>
```
Comment (TheJJ) [inline libopenage/renderer/text_renderer.cpp]: there's no opengl call in that file, so no need for the epoxy header.

Comment (jprashanth) [inline libopenage/renderer/text_renderer.cpp]: glColor4f is being used in the render method.

File: libopenage/engine.cpp (hunk 4 above)
Comment (jprashanth) [inline libopenage/engine.cpp]: Yes, just demonstrating the usage.

File: libopenage/renderer/text_renderer.cpp (hunk 5 above)
Comment (TheJJ) [inline libopenage/renderer/text_renderer.cpp]: you're right, sorry.

**Comment by jprashanth** [comment]
Completed the required changes.

**Comment by TheJJ** [comment]
Thanks! Cool stuff.

**Pull request merged**